core/rfb.js

```diff
@@ -1805,19 +1827,34 @@ RFB.encodingHandlers = {
             this._FBU.lines = this._FBU.height;
         }
 
-        this._FBU.bytes = this._FBU.width * 4;  // at least a line
+        var pixelSize = this._fb_depth == 8 ? 1 : 4;
```

Comment:
it feels hacky to hardcode this.  We should calculate it from by dividing depth by 8, and then cache if necessary (or change depth to be bytes, not bits).

Reply:
But depth 24 bits has a pixel size of 32 bits, so we have to have special code for this.

Reply:
We used to have a difference between depth and pixel size, and we passed that in explicitly, so the normal combo was `(4, 3)` (4 bytes per pixel, 3 bytes of depth).  I'd prefer explicitly passing those as separate arguments.

Reply:
See above. They're not independent and having both as arguments give the illusion that they are.